### ISSUE TITLE
[Feature Request]: Log Service

### ISSUE THREAD

**Issue by lni** [matrixorigin/matrixone#2757]
### Is there an existing issue for enhancement?

- [X] I have checked the existing issues.

### What would you like to be added ?

```markdown
A LogService running as an independent service for efficiently storing transactional logs. 

This issue is for the LogService itself, it doesn't cover a client library used for accessing the Log Service. High availability of the Log Service will be covered in a separate issue as well.
```

### Why is this needed ?

This component is required in the new architecture. It helps to realise the separation of storage and compute.

  

### Additional information

More details of 
https://github.com/matrixorigin/docs/tree/main/api/logservice

**Comment by lni** [comment]
Based on previous discussions, the following design decisions & assumptions have been made - 

1. Each DN should get multiple log service clients to append LogRecords into the Log Service in parallel. 
2. Each such LogRecord should contain transaction data from multiple transactions (group commit). Best write throughput can be achieved when appending LogRecords at 16Bytes each when there are hundreds of concurrent clients keep appending LogRecords. 
3. An uint64 Lsn value will be assigned & returned to the caller on the successful return of the Append() method. There is no guarantee that such returned Lsn values will be continuous as internal commands which are not visible to users will also be assign Lsn values in the same Lsn space. 
4. Errors are inevitable in any distributed system - e.g. errors caused by timeouts and connectivity issues to just name a few. It is caller's responsibility to properly handle such errors and retry requests when necessary.
